src/de/espend/idea/laravel/stub/processor/BladeCustomDirectivesVisitor.java

```diff
@@ -0,0 +1,69 @@
+package de.espend.idea.laravel.stub.processor;
+
+import com.intellij.openapi.util.Pair;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiRecursiveElementVisitor;
+import com.jetbrains.php.lang.psi.elements.Method;
+import com.jetbrains.php.lang.psi.elements.MethodReference;
+import com.jetbrains.php.lang.psi.elements.StringLiteralExpression;
+import fr.adrienbrault.idea.symfony2plugin.Symfony2InterfacesUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class BladeCustomDirectivesVisitor extends PsiRecursiveElementVisitor {
+
+    private Set<String> availableClasses = new HashSet<>();
+
+    private Consumer<Pair<PsiElement, String>> consumer;
+
+    public BladeCustomDirectivesVisitor(@NotNull Consumer<Pair<PsiElement, String>> consumer) {
+        this.consumer = consumer;
+
+        availableClasses.add("\\Illuminate\\Support\\Facades\\Blade");
+        availableClasses.add("\\Blade");
+        availableClasses.add("Blade");
+    }
+
+    @Override
+    public void visitElement(PsiElement element) {
+
+        try {
```

Comment:
I added try{ only for finally section. Without finally I had to replace each { return; } to { super.visitElement(element); return; } :) Ofc I can change it if you don't like try{ here.